nautobot/utilities/views.py

```diff
@@ -151,3 +186,1062 @@ def get_return_url(self, request, obj=None):
+    table: The django-tables2 Table used to render the objects list
+    template_name: The name of the template
+    """
+
+    object_list_action_buttons = ("add", "import", "export")
+
+    def __init__(self, *args, **kwargs):
+        self.routes = self.define_routes()
+
+    def define_routes(self):
+        return super().define_routes() + [
+            Route(
+                url=r"^{prefix}/$",
+                mapping={
+                    "get": "handle_object_list_get",
+                },
+                name="{basename}_list",
+                detail=False,
+                initkwargs={"suffix": "List"},
+            ),
+        ]
+
+    def queryset_to_yaml(self):
+        """
+        Export the queryset of objects as concatenated YAML documents.
+        """
+        yaml_data = [obj.to_yaml() for obj in self.queryset]
+
+        return "---\n".join(yaml_data)
+
+    def queryset_to_csv(self):
+        """
+        Export the queryset of objects as comma-separated value (CSV), using the model's to_csv() method.
+        """
+        csv_data = []
+        custom_fields = []
+
+        # Start with the column headers
+        headers = self.queryset.model.csv_headers.copy()
+
+        # Add custom field headers, if any
+        if hasattr(self.queryset.model, "_custom_field_data"):
+            for custom_field in CustomField.objects.get_for_model(self.queryset.model):
+                headers.append(custom_field.name)
+                custom_fields.append(custom_field.name)
+
+        csv_data.append(",".join(headers))
+
+        # Iterate through the queryset appending each object
+        for obj in self.queryset:
+            data = obj.to_csv()
+
+            for custom_field in custom_fields:
+                data += (obj.cf.get(custom_field, ""),)
+
+            csv_data.append(csv_format(data))
+
+        return "\n".join(csv_data)
+
+    def handle_object_list_get(self, request):
+        model = self.queryset.model
+        content_type = ContentType.objects.get_for_model(model)
+
+        self.queryset = self.filterset(request.GET, self.queryset).qs
+
+        # Check for export template rendering
+        if request.GET.get("export"):
+            et = get_object_or_404(
+                ExportTemplate,
+                content_type=content_type,
+                name=request.GET.get("export"),
+            )
+            try:
+                return et.render_to_response(self.queryset)
+            except Exception as e:
+                messages.error(
+                    request,
+                    "There was an error rendering the selected export template ({}): {}".format(et.name, e),
+                )
+
+        # Check for YAML export support
+        elif "export" in request.GET and hasattr(model, "to_yaml"):
+            response = HttpResponse(self.queryset_to_yaml(), content_type="text/yaml")
+            filename = "nautobot_{}.yaml".format(self.queryset.model._meta.verbose_name_plural)
```

Comment:
I know this is copy-pasta from the existing generic view classes, but can we please change all instances of `"....".format()` to f-strings?

Reply:
Should I change all the .format() to f-strings in this PR or should I make a separate one?

Reply:
For now in this PR just convert to f-strings and we should have it in a style guide to encourage their use. I don't know if it makes sense to have that type of style change be it's own separate housekeeping issue unless there is a clear performance/forced style checking in the future.